### PR TITLE
IGNITE-19693 getAll does not preserve order and does not return nulls for missing keys

### PR DESCRIPTION
**Pull request by tkalkirill**
https://issues.apache.org/jira/browse/IGNITE-19693

### CODE REVIEW

File: modules/api/src/main/java/org/apache/ignite/table/RecordView.java [hunk 1]
```diff
@@ -55,18 +56,20 @@ public interface RecordView<R> extends DataStreamerTarget<R> {
...
      *
      * @param tx      Transaction or {@code null} to auto-commit.
      * @param keyRecs Records with the key columns set. The records cannot be {@code null}.
-     * @return Future that represents the pending completion of the operation.
+     * @return Future that will return records with all columns filled from the table. The order of collection elements is
+     *      guaranteed to be the same as the order of {@code keyRecs}. If a record does not exist, the
+     *      element at the corresponding index of the resulting collection is {@code null}.
      */
-    @NotNull CompletableFuture<Collection<R>> getAllAsync(@Nullable Transaction tx, @NotNull Collection<R> keyRecs);
+    CompletableFuture<List<R>> getAllAsync(@Nullable Transaction tx, Collection<R> keyRecs);
```
Comment (rpuch) [inline modules/api/src/main/java/org/apache/ignite/table/RecordView.java:72]: Same question about the type change

File: modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientKeyValueView.java [hunk 2]
```diff
@@ -464,11 +464,12 @@ private HashMap<K, V> readGetAllResponse(ClientSchema schema, ClientMessageUnpac
 
         try {
             for (int i = 0; i < cnt; i++) {
-                in.unpackBoolean(); // TODO: Optimize (IGNITE-16022).
-
-                var tupleReader = new BinaryTupleReader(schema.columns().length, in.readBinaryUnsafe());
-                var reader = new ClientMarshallerReader(tupleReader);
-                res.put((K) keyMarsh.readObject(reader, null), (V) valMarsh.readObject(reader, null));
+                // TODO: Optimize (IGNITE-16022).
+                if (in.unpackBoolean()) {
```
Comment (rpuch) [inline modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientKeyValueView.java:468]: Does this boolean mean that the following value is not null?

File: modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientRecordBinaryView.java [hunk 3]
```diff
@@ -76,27 +78,26 @@ public Tuple get(@Nullable Transaction tx, @NotNull Tuple keyRec) {
                 ClientTupleSerializer.getPartitionAwarenessProvider(tx, keyRec));
     }
 
-    /** {@inheritDoc} */
     @Override
-    public Collection<Tuple> getAll(@Nullable Transaction tx, @NotNull Collection<Tuple> keyRecs) {
+    public List<Tuple> getAll(@Nullable Transaction tx, Collection<Tuple> keyRecs) {
```
Comment (rpuch) [inline modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientRecordBinaryView.java:82]: List?

File: modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientRecordView.java [hunk 4]
```diff
@@ -76,27 +78,26 @@ public R get(@Nullable Transaction tx, @NotNull R keyRec) {
                 ClientTupleSerializer.getPartitionAwarenessProvider(tx, ser.mapper(), keyRec));
     }
 
-    /** {@inheritDoc} */
     @Override
-    public Collection<R> getAll(@Nullable Transaction tx, @NotNull Collection<R> keyRecs) {
+    public List<R> getAll(@Nullable Transaction tx, Collection<R> keyRecs) {
```
Comment (rpuch) [inline modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientRecordView.java:82]: List?

File: modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientKeyValueView.java (hunk 2 above)
Comment (tkalkirill) [reply modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientKeyValueView.java:468]: I think yes, this part was done by @ptupitsyn

**Review by rpuch — APPROVED** [review]

**Pull request merged**